### DISCUSSION TITLE
jotai - react-native - async storage

### DISCUSSION THREAD

**Discussion by smontlouis** [pmndrs/jotai#1713] [Q&A] [answered]
I'm not able to get jotai to work properly with react-native with async-storage.

First I used Suspense with atomWithStorage, but then for whatever reason, app screen gets stuck on splashscreen.

Then I read you can to use loadableAtoms with delayInit:false on atomWithStorage. 
It’s still not working because loadable has state attribute initialized to hasData (instead of loading) so I can’t wait for the data to finish loading even in this way.

If I rollback to jotai 1.10.0, loadable atoms seems to work properly, but then for some users, for whatever reason, data is not persisted (I believe `hasData` to be always true for some users)

If I use last jotai version 1.13.1, loadable atoms has state initialized to hasData.

Ideally I would like a solid solution using suspense, but if it's not possible I would like the loadable to work properly.

**Comment by dai-shi** [comment]
In the next version, `delayInit: true` by default.
So, you want to use `delayInit: true` now.
If it doesn't fill your needs, it's time to create your own atom util instead of using atomWithStorage.
(I think we should improve the persistence guide: https://jotai.org/docs/guides/persistence )

**Reply by smontlouis** [reply]
Thank you. 
But even with `delayInit: true`, loadable state is always to `hasData` (instead of `loading` then `hasData`) when loading the app.

**Reply by dai-shi** [reply]
yeah, I mean `delayInit: true` is to avoid suspense, so there's no point of using `loadable`. If you need the suspense on load, creating own helper would be good. I can help guiding you and you can help us improving the docs? 😁

**Reply by smontlouis** [reply]
Yes please! I misunderstood the docs. 

Yes now I'll be glad to help writing the docs now that chat GPT can improve my writing 😅

**Comment by smontlouis** [comment, marked as answer]
It seems that loadable is working using jotai v2 api !

**Reply by dai-shi** [reply]
That's great to hear.
